### PR TITLE
feat(DCGO): add presence

### PR DESCRIPTION
**Pull request by pudidev**
## Description 
Enter the instant server ranking with DiscordGO.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/D/DCGO/presence.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		presenceData.details = "Bir ayar görüntülüyor:";
+		presenceData.state = "Bağlantılar";
+	} else if (window.location.pathname.toLowerCase() === "/help") {
+		presenceData.details = "Bir sayfa görüntülüyor:";
+		presenceData.state = "Yardım Sayfası";
+	} else if (window.location.pathname.toLowerCase().includes("/profile")) {
+		presenceData.details = "Bir kullanıcı profili görüntülüyor:";
+		presenceData.state = `${
+			document.querySelector(
+				"body > div.main-wrapper > div.main-content > div > div > div > div.col-lg-12 > div > div.card-body.d-block.pt-4.text-center.position-relative > h4"
```
Comment (theusaf) [inline websites/D/DCGO/presence.ts:37]: Try to simplify long, specific selectors like these. For example:
```suggestion
				"h4"
```

File: websites/D/DCGO/tsconfig.json
Comment (theusaf) [inline websites/D/DCGO/tsconfig.json:1]: This file should be deleted.

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — CHANGES_REQUESTED** [review]
PR title does not follow commit conventions.

**Review by turkwr — COMMENTED** [review]
changed

File: websites/D/DCGO/metadata.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		"name": "Türker"
+	},
+	"service": "DCGO",
+	"description": {
+		"en": "See instant server ranking with DiscordGO.",
+		"tr": "DiscordGO ile anlık sunucu sıralamasını görün."
+	},
+	"url": "discordgo.net",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/hDOZmml.png",
```
Comment (DarkVillager) [inline websites/D/DCGO/metadata.json:14]: Logo must be 512x512.

File: websites/D/DCGO/presence.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+const presence = new Presence({
+	clientId: "1075115687888044072",
+});
+
+presence.on("UpdateData", () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/81z9xFR.png",
```
Comment (DarkVillager) [inline websites/D/DCGO/presence.ts:7]: Same here. Log must be 512x512.

File: websites/D/DCGO/presence.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	};
+	if (window.location.pathname.toLowerCase() === "/") {
```
[the remark this answers is not in the document]
Comment (DarkVillager) [reply websites/D/DCGO/presence.ts:9]: ^ @turkwr acutally make requested changes or your PR will most likely get closed.

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by turkwr** [comment]
@DarkVillager

File: websites/D/DCGO/presence.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (DarkVillager) [reply websites/D/DCGO/presence.ts:9]: > ^ @turkwr acutally make requested changes or your PR will most likely get closed.

Again.

**Review by theusaf — CHANGES_REQUESTED** [review]
The `tsconfig.json` file should be removed.

**Comment by turkwr** [comment]
@theusaf

**Review by Bas950 — CHANGES_REQUESTED** [review]
Use `document` instead of `window`, it also might be easier for you to add a deconstructed pathname variable and just use that instead.

For example: `const { pathname } = document.location;`
and then using that in your `if`s instead.

File: websites/D/DCGO/presence.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (DarkVillager) [reply websites/D/DCGO/presence.ts:9]: Again.

File: websites/D/DCGO/presence.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const presence = new Presence({
+	clientId: "1075115687888044072",
+});
+
+presence.on("UpdateData", () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/NPVqgsx.png",
+	};
+	if (document.location.pathname.toLowerCase() === "/") {
```
Comment (theusaf) [inline websites/D/DCGO/presence.ts:9]: As `document.location.pathname` is used many times in this presence, destructure this property for simplicity.
```suggestion
	const { pathname } = document.location;
	if (pathname.toLowerCase() === "/") {
```

File: websites/D/DCGO/presence.ts (hunk 1 above)
Comment (theusaf) [reply websites/D/DCGO/presence.ts:37]: As previously requested, websites change often, so selectors that can be made less specific are better.

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
@mepuddly maybe make requested changes before this PR gets closed.

File: websites/D/DCGO/presence.ts (hunk 5 above)
Comment (mepuddly) [reply websites/D/DCGO/presence.ts:9]: okey

Comment (mepuddly) [reply websites/D/DCGO/presence.ts]: i changed

Comment (mepuddly) [reply websites/D/DCGO/presence.ts]: @theusaf

File: websites/D/DCGO/presence.ts (hunk 1 above)
Comment (mepuddly) [reply websites/D/DCGO/presence.ts:38]: When it will be updated, I will update it myself.

**Comment by mepuddly** [comment]
@DarkVillager

**Comment by mepuddly** [comment]
I did as you said, but this time it didn't pass the check. What is the reason?

File: websites/D/DCGO/presence.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (mepuddly) [reply websites/D/DCGO/presence.ts]: changed

File: websites/D/DCGO/presence.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
+const presence = new Presence({
+	clientId: "1075115687888044072",
+});
+
+presence.on("UpdateData", () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/NPVqgsx.png",
+	};
+
+	const { pathname } = document.location;
```
Comment (DarkVillager) [inline websites/D/DCGO/presence.ts:10]:
```suggestion
	const { pathname } = document.location,
	pathnameLowercase = pathname.toLowerCase();
```

Comment (DarkVillager) [inline websites/D/DCGO/presence.ts:10]:
```suggestion
	},
	 { pathname } = document.location,
	pathnameLowercase = pathname.toLowerCase();
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/D/DCGO/presence.ts (hunk 6 above)
Comment (mepuddly) [reply websites/D/DCGO/presence.ts:10]: i updated

**Comment by mepuddly** [comment]
@DarkVillager i changed

File: websites/D/DCGO/presence.ts (hunk 6 above)
Comment (DarkVillager) [reply websites/D/DCGO/presence.ts:10]: Don't forget to add this otherwise your changes won't work.

Comment (mepuddly) [reply websites/D/DCGO/presence.ts:10]: I don't know what you're talking about.

Comment (mepuddly) [reply websites/D/DCGO/presence.ts:10]: Oh, okay, I get it.

File: websites/D/DCGO/presence.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
+const presence = new Presence({
+	clientId: "1075115687888044072",
+});
+
+presence.on("UpdateData", () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/NPVqgsx.png",
+	},
+	 { pathname } = document.location,
+	pathnameLowercase = pathname.toLowerCase();
```
Comment (DarkVillager) [inline websites/D/DCGO/presence.ts:10]:
```suggestion
	pathnameLowerCase = pathname.toLowerCase();
```

File: websites/D/DCGO/presence.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+	if (pathnameLowercase  === "/") {
```
Comment (DarkVillager) [inline websites/D/DCGO/presence.ts:11]:
```suggestion
	if (pathnameLowerCase === "/") {
```

File: websites/D/DCGO/presence.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+		presenceData.details = "Bir sayfa görüntülüyor:";
+		presenceData.state = "Anasayfa";
+	} else if (pathname.toLowerCase() === "/team") {
```
Comment (DarkVillager) [inline websites/D/DCGO/presence.ts:14]:
```suggestion
	} else if (pathnameLowerCase === "/team") {
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
image
??

**Comment by mepuddly** [comment]
our site is currently under maintenance

**Comment by mepuddly** [comment]
@Bas950 website online.

**Review by Bas950 — APPROVED** [review]
Not a whole lot to the website... but ok

**Comment by Bas950** [comment]
Actually why is it named DCGO and not DiscordGo, from the website I only see DiscordGo, no where DCGO...

**Comment by mepuddly** [comment]
> Actually why is it named DCGO and not DiscordGo, from the website I only see DiscordGo, no where DCGO...

Because Discord does not allow the name when creating the application.

**Comment by mepuddly** [comment]
> Actually why is it named DCGO and not DiscordGo, from the website I only see DiscordGo, no where DCGO...

Because Discord does not allow the name when creating the application.

**Review by theusaf — APPROVED** [review]

**Comment by DarkVillager** [comment]
> 

You can?
image

**Review by theusaf — CHANGES_REQUESTED** [review]
Also, make sure to resolve CI issues. (run `yarn format`)

**Comment by pudidev** [comment]
I will try it right away

**Comment by pudidev** [comment]
image

**Review by theusaf — CHANGES_REQUESTED** [review]
The code is still not formatted properly. Also, be sure to remove the changes to presences unrelated to this PR (For example, using `git checkout`, `git rebase`, etc.). If you need help with this, you can reach out on Discord.

**Comment by pudidev** [comment]
> The code is still not formatted properly. Also, be sure to remove the changes to presences unrelated to this PR (For example, using `git checkout`, `git rebase`, etc.). If you need help with this, you can reach out on Discord.

I don't quite understand.

**Comment by theusaf** [comment]
> > The code is still not formatted properly. Also, be sure to remove the changes to presences unrelated to this PR (For example, using `git checkout`, `git rebase`, etc.). If you need help with this, you can reach out on Discord.
> 
> I don't quite understand.

If you take a look at the "files" tab on this PR, you will see files that are not related to DCGO. These changes should be reverted (Merging/Rebasing the latest commits may help with this).
Looking at the code for this presence also displays code style issues, such as incorrect indentation. Use `npm run format` to format it, but make sure you only commit the files related to this PR.

**Comment by pudidev** [comment]
I will update when I can.

**Comment by theusaf** [comment]
Closing for inactivity. You may open a new PR after resolving the issues mentioned in this one.

**Pull request closed without merging**